src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
```

Comment:
PullProofAI

Consider adding error handling for any potential exceptions that may occur during the construction of `JwtStrategy`.

```diff
+ try {
+     super({
+       secretOrKey: Env.JWT.AUTH_TOKEN,
+       jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+     });
+ } catch (error) {
+     // Handle any errors that occur during strategy construction
+     throw new Error('Error constructing JwtStrategy: ' + error.message);
+ }
```



<!-- This is an auto-generated comment by PullProofAI -->